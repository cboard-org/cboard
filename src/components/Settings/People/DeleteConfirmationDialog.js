import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import { FormattedMessage } from 'react-intl';
import messages from './People.messages';

import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  LinearProgress
} from '@material-ui/core';

const propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func,
  handleDeleteConfirmed: PropTypes.func,
  isDeletingAccount: PropTypes.bool,
  errorDeletingAccount: PropTypes.bool
};

const defaultProps = {};

const DeleteConfirmationDialog = ({
  open,
  handleClose,
  handleDeleteConfirmed,
  isDeletingAccount,
  errorDeletingAccount
}) => {
  const DELETE_ACCOUNT = 'delete-account';
  const [confirmationText, setConfirmationText] = useState('');

  const handleConfirmationChange = e => {
    setConfirmationText(e.target.value);
  };

  const handleDialogClose = () => {
    setConfirmationText('');
    handleClose();
  };

  return (
    <Dialog
      open={open}
      onClose={handleClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">
        {<FormattedMessage {...messages.deleteAccountPrimary} />}
      </DialogTitle>
      <DialogContent>
        {errorDeletingAccount ? (
          <DialogContentText id="alert-dialog-description">
            <FormattedMessage {...messages.errorDeletingAccount} />
          </DialogContentText>
        ) : (
          <DialogContentText id="alert-dialog-description">
            <FormattedMessage {...messages.deleteAccountConfirmation} />
          </DialogContentText>
        )}
        <TextField
          autoFocus={true}
          fullWidth={true}
          label={
            <FormattedMessage
              {...messages.deleteAccountFinal}
              values={{ deleteAccount: DELETE_ACCOUNT }}
            />
          }
          onChange={handleConfirmationChange}
        />
      </DialogContent>
      {!isDeletingAccount && (
        <DialogActions>
          <Button
            variant="outlined"
            color="secondary"
            className={'delete_button'}
            disabled={confirmationText !== DELETE_ACCOUNT}
            onClick={handleDeleteConfirmed}
          >
            {<FormattedMessage {...messages.deleteAccountPrimary} />}
          </Button>
          <Button variant="outlined" onClick={handleDialogClose} autoFocus>
            {<FormattedMessage {...messages.cancelDeleteAccount} />}
          </Button>
        </DialogActions>
      )}
      {isDeletingAccount && <LinearProgress color="secondary" />}
    </Dialog>
  );
};

DeleteConfirmationDialog.propTypes = propTypes;
DeleteConfirmationDialog.defaultProps = defaultProps;

export default DeleteConfirmationDialog;
