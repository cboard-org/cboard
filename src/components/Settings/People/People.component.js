import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';

import FullScreenDialog from '../../UI/FullScreenDialog';
import messages from './People.messages';

const propTypes = {
  /**
   * Callback fired when clicking the back button
   */
  onClose: PropTypes.func
};

const People = ({ onClose }) => (
  <div className="People">
    <FullScreenDialog
      open
      title={<FormattedMessage {...messages.people} />}
      onClose={onClose}
    >
      <Paper>
        <Button>Sign In</Button>
      </Paper>
    </FullScreenDialog>
  </div>
);

People.propTypes = propTypes;

export default People;
