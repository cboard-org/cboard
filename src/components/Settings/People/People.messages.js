import { defineMessages } from 'react-intl';

export default defineMessages({
  people: {
    id: 'cboard.components.Settings.People.people',
    defaultMessage: 'People'
  }
});
