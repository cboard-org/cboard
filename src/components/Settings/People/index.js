export { default } from './People.container';
