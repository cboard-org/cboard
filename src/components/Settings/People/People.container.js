import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import People from './People.component';

export class PeopleContainer extends PureComponent {
  static propTypes = {
    history: PropTypes.object.isRequired
  };

  render() {
    const { history } = this.props;

    return <People onClose={history.goBack} />;
  }
}

const mapStateToProps = state => ({});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(PeopleContainer);
