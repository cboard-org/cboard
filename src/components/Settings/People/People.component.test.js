import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import People from './People.component';

jest.mock('./People.messages', () => {
  return {
    people: {
      id: 'cboard.components.Settings.People.people',
      defaultMessage: 'People'
    },
    name: {
      id: 'cboard.components.Settings.People.name',
      defaultMessage: 'Name'
    },
    nameSecondary: {
      id: 'cboard.components.Settings.People.nameSecondary',
      defaultMessage: 'Update your user name'
    },
    email: {
      id: 'cboard.components.Settings.People.email',
      defaultMessage: 'Email'
    },
    emailSecondary: {
      id: 'cboard.components.Settings.People.emailSecondary',
      defaultMessage: 'Update your user email'
    },
    language: {
      id: 'cboard.components.Settings.People.language',
      defaultMessage: 'Language'
    },
    languageSecondary: {
      id: 'cboard.components.Settings.People.languageSecondary',
      defaultMessage: 'Let us know what is your language'
    },
    birthdate: {
      id: 'cboard.components.Settings.People.birthdate',
      defaultMessage: 'Birth Date'
    },
    birthdateSecondary: {
      id: 'cboard.components.Settings.People.birthdateSecondary',
      defaultMessage: 'Update your birth date'
    },
    location: {
      id: 'cboard.components.Settings.People.location',
      defaultMessage: 'Location'
    },
    logout: {
      id: 'cboard.components.Settings.People.logout',
      defaultMessage: 'Logout'
    }
  };
});

describe('People tests', () => {
  const onChangePeople = jest.fn();
  const onSubmitPeople = jest.fn();
  const onClose = jest.fn();
  const logout = jest.fn();
  test('default renderer', () => {
    shallowMatchSnapshot(
      <People
        isLogged={false}
        onClose={onClose}
        onChangePeople={onChangePeople}
        onSubmitPeople={onSubmitPeople}
        logout={logout}
      />
    );
  });
});
