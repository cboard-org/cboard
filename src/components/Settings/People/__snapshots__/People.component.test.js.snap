// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People tests default renderer 1`] = `
<div
  className="People"
>
  <WithStyles(FullScreenDialog)
    onClose={[Function]}
    open={true}
    title={
      <FormattedMessage
        defaultMessage="People"
        id="cboard.components.Settings.People.people"
        values={Object {}}
      />
    }
  >
    <WithStyles(Paper)>
      <WithStyles(Button)>
        Sign In
      </WithStyles(Button)>
    </WithStyles(Paper)>
  </WithStyles(FullScreenDialog)>
</div>
`;
